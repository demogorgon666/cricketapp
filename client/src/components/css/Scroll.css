.menu-item {
    padding: 0 40px;
    margin: 10px 20px;
    user-select: none;
    cursor: pointer;
    border: none;
  }
  .menu-item-wrapper.active {
    border: px blue solid;
  }
  .menu-item.active {
    border: 1px blue solid;
  }
   
  .scroll-menu-arrow {
    padding: 30px;
    cursor: pointer;
  }