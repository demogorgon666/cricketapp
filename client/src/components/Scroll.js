import React, { Component } from 'react';
import '../components/css/Scroll.css'

// list of items
export const list = [
    { date: 'SEP 19' },
    { date: 'SEP 20' },
    { date: 'SEP 21' },
    { date: 'SEP 22' },
    { date: 'SEP 23' },
    { date: 'SEP 24' },
    { date: 'SEP 25' },
    { date: 'SEP 26' },
    { date: 'SEP 27' },
    { date: 'SEP 28' },
    { date: 'SEP 29' },
    { date: 'SEP 30' }
];

// One item component
// selected prop will be passed
export const MenuItem = ({ text, selected }) => {
    return <div
        className={`menu-item ${selected ? 'active' : ''}`}
    >{text}</div>;
};

// All items component
// Important! add unique key
export const Menu = (list, selected) =>
    list.map(el => {
        const { date } = el;

        return <MenuItem text={date} key={date} selected={selected} />;
    });


export const Arrow = ({ text, className }) => {
    return (
        <div
            className={className}
        >{text}</div>
    );
};


export const ArrowLeft = Arrow({ text: '<', className: 'arrow-prev' });
export const ArrowRight = Arrow({ text: '>', className: 'arrow-next' });

export const selected = 'SEP 26';