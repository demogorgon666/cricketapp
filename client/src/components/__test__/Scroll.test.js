import React from "react";
import { shallow } from "enzyme";
import { Scroll } from "../Scroll";

const scroller = jest.fn();
var getMatchesDate = jest.fn();
var date = [];
const wrapper = shallow(
  <Scroll scroller={scroller} date={date} getMatchesDate={getMatchesDate} />
);

describe("Testing of Scroll Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //   it("should have 2 div tag", () => {
  //     expect(wrapper.find("div").length).toBe(7);
  //   });
});
