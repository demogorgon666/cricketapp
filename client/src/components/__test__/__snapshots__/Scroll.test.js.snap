// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of Scroll Component should render the component 1`] = `<div />`;
